### PR TITLE
Fix TS types

### LINKED ISSUE #304: Multipart: Boundary not found
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.24.1

### Plugin version

5.2.0

### Node.js version

16.5.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Ubuntu 20.04

### Description

When the file limit size is set, then headers are a mandatory prop. The code should be something like:

```js
await req.file({
    headers: { "content-type": "multipart/form-data" },
    limits: {
        fileSize: (1024 * 1024) * 55
    }
})
```

The issue begins with passing those content-type at the request (as they are mandatory now), then the server responds with error 500 and the message "Multipart: Boundary not found".

### Steps to Reproduce

1 - Make a file upload;
2 - Set some limit to the upload throught req.file(PROPS);
3 - Verify that the headers are mandatory now;
4 - Pass the respective header throught request;
5 - Try to do the upload;
6 - Observe that the server responds with error 500 with the message "Multipart: Boundary not found".

### Expected Behavior

The upload happens normally respecting the file size limit.

### PR DESCRIPTION
**Pull request by kibertoad**
fixes #304

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by kibertoad** [comment]
@climba03003 Technically busboy types are correct, it does need that parameter for functioning. However, fastify-multipart types were wrong, because we were not relying on user to pass that parameter, we were retrieving it from request directly.

File: index.d.ts [hunk 1]
```diff
@@ -90,6 +90,16 @@ export interface FastifyMultipartBaseOptions {
...
+    /**
+     * Detect if a Part is a file.
+     *
+     * By default a file is detected if contentType
+     * is application/octet-stream or fileName is not
+     * undefined.
+     *
+     * Modify this to handle e.g. Blobs.
+     */
+    isPartAFile?: (fieldName: string | undefined, contentType: string | undefined, fileName: string | undefined) => boolean;
```
Comment (kibertoad) [inline index.d.ts:102]: This part of busboy types was missing

**Comment by climba03003** [comment]
> @climba03003 Technically busboy types are correct, it does need that parameter for functioning. However, fastify-multipart types were wrong, because we were not relying on user to pass that parameter, we were retrieving it from request directly.

I will use `Omit<BusboyConfig, 'headers'>` as we handle this already. 
`Partial` also allow the users to pass the `headers` which may lead to the same problem again.

**Comment by kibertoad** [comment]
@climba03003 Good point. Fixed.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by rhadamez** [comment]
You guys are too fast, thats how a real framework should happen. Thanks.

**Pull request merged**